src/TableColumn/FilterModel/TypeDatetime.php

```diff
@@ -92,7 +92,7 @@ public function setConditionForModel($m)
                     break;
                 case '!=':
                 case '=':
-                    $d1 = $this->getDatetime($filter['value'])->setTime(0, 0, 0);
+                    $d1 = clone $this->getDatetime($filter['value'])->setTime(0, 0, 0);
                     $d2 = $this->getDatetime($filter['value'])->setTime(23, 59, 59);
                     if ($d2 >= $d1) {
```

Comment:
Reading through this part if the code, to me it seems this condition is always true. I cant figure how the else part could possibly be reached. As the same parameter is passed to getDateTime(), $d2 will always be greater then $d1.